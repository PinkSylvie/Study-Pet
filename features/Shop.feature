Feature: Shop
  Verify that all relevant features pertaining to the shop work.

  Background:
    Given User at Home Page
    When User Clicks The Navigate to Shop Page Button

  Scenario: Navigate To The Bath Page
    Then User At Shop Page

  Scenario Outline: Filtering
    When User Filters by "<Filter>"
    Then "<Filter>" Items Show

    Examples:
      | Filter |
      | All |
      | Food |
      | Toys |
      | Clothes |

    Scenario Outline: Buying
      When User Buys "<Item>"
      Then Money Left is "<Balance>"

      Examples:
        | Item | Balance |
        | Orange Juice | 490 |
        | Apple Juice | 480 |
        | Banana Juice | 480 |
        | Ball | 450 |
        | Los Diablos Ski Mask | 460 |
        | All | 360 |

    Scenario: Resetting Money
      When User Buys "All"
      And User Resets Money
      Then Money Left is "999"
