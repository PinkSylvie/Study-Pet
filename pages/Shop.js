import { Ionicons } from "@expo/vector-icons";
import React, { useState } from "react";
import {
  View,
  Text,
  TouchableOpacity,
  FlatList,
  Image,
  Pressable,
} from "react-native";
import { StyleSheet } from "react-native";
import { useFonts } from "expo-font";
import { allShopItems, allShopCategories } from "../utils/shopAssets";

const Shop = ({ studyPoints, setStudyPoints }) => {
  const [dummy] = useFonts({
    // "Jua-Regular": require("../assets/fonts/Jua-Regular.ttf"),
  });
  const [category, setCategory] = useState("All");
  const [items, setItems] = useState(allShopItems);

  const buyItem = (item) => {
    if (studyPoints >= item.cost) {
      setStudyPoints(studyPoints - item.cost);
      // Perform the logic to give the user the item
      console.log(`You bought ${item.name}`);
    } else {
      console.log(`Not enough study points to buy ${item.name}`);
    }
  };

  const renderItem = ({ item }) => (
    <TouchableOpacity id={`shop-item-${item.id}`} onPress={() => buyItem(item)}>
      <View style={styles.item}>
        <Image source={{ uri: item.image }} style={styles.image} />
        <View
          style={{
            flex: 1,
            justifyContent: "space-between",
            flexDirection: "row",
          }}
        >
          <Text id={`shop-item-${item.id}-name`} style={styles.itemName}>{item.name}</Text>
          <Text id={`shop-item-${item.id}-cost`} style={styles.itemCost}>{item.cost}</Text>
        </View>
        <Ionicons
          name="logo-bitcoin"
          size={24}
          color="black"
          style={styles.itemIcon}
        />
      </View>
    </TouchableOpacity>
  );

  const renderCategory = ({ item }) => (
    <View style={styles.category}>
      <TouchableOpacity
        id={`shop-category-${item}`}
        onPress={() => setCategory(item)}
        styles={styles.category}
      >
        <Text style={styles.categoryText}>{item}</Text>
      </TouchableOpacity>
    </View>
  );

  return (
    <View style={styles.shopContainer}>
      <View style={{ flexDirection: "row" }}>
        <View
          style={{
            width: 60,
            flexDirection: "row",
            justifyContent: "space-evenly",
            marginLeft: 10,
          }}
        >
          <Text
            id={`shop-study-points`}
            style={{
              fontSize: 26,
              fontFamily: "Jua-Regular",
              alignSelf: "center",
            }}
          >
            {studyPoints}
          </Text>
          <Ionicons
            name="logo-bitcoin"
            size={24}
            color="black"
            style={{ alignSelf: "center" }}
          />
        </View>
        <FlatList
          style={{
            alignSelf: "center",
            margin: 5,
            padding: 5,
            flexDirection: "row",
          }}
          data={allShopCategories}
          renderItem={renderCategory}
          horizontal={true}
        />
      </View>
      <FlatList
        data={items.filter(
          (item) => item.category === category || category === "All"
        )}
        renderItem={renderItem}
      />
      <Pressable id={`shop-reset-points-button`} style={{ padding: 10 }} onPress={() => setStudyPoints(999)}>
        <Text style={{ fontSize: 26, fontFamily: "Jua-Regular" }}>
          Reset points
        </Text>
      </Pressable>
    </View>
  );
};

// Stylesheet for item, shows item, name, cost and category with picture of item
const styles = StyleSheet.create({
  shopContainer: {
    flex: 1,
    backgroundColor: "#89CFF0",
    paddingBottom: 25,
  },
  item: {
    padding: 5,
    flexDirection: "row",
    backgroundColor: "#ffffff",
    margin: 5,
    borderRadius: 10,
  },
  image: {
    width: 50,
    height: 50,
    alignSelf: "center",
  },
  itemName: {
    fontSize: 16,
    fontWeight: "bold",
    alignSelf: "center",
    marginLeft: 10,
  },
  itemCost: {
    fontSize: 16,
    alignSelf: "center",
  },
  itemIcon: {
    alignSelf: "center",
  },
  category: {
    padding: 5,
    backgroundColor: "#98FB98",
    margin: 5,
    borderRadius: 10,
  },
  categoryText: {
    fontSize: 26,
    fontFamily: "Jua-Regular",
  },
});
export default Shop;
