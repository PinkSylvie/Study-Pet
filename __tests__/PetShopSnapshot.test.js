import React from 'react';
import '@testing-library/jest-native/extend-expect'; // if you use @testing-library/react-native
import { render } from '@testing-library/react-native';
import Shop from '../pages/Shop.js'; // Adjust the path as necessary

// Mocks for specific libraries
jest.mock('expo-font', () => ({
    useFonts: jest.fn().mockReturnValue([true]), // Simulate fonts are loaded
}));

describe('Shop Component UI', () => {
    it('renders correctly and matches snapshot', () => {
        const { toJSON } = render(<Shop studyPoints={100} setStudyPoints={() => { }} />);
        expect(toJSON()).toMatchSnapshot();
    });
});
