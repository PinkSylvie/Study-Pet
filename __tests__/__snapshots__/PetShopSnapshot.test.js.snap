// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop Component UI renders correctly and matches snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#89CFF0",
      "flex": 1,
      "paddingBottom": 25,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginLeft": 10,
          "width": 60,
        }
      }
    >
      <Text
        id="shop-study-points"
        style={
          {
            "alignSelf": "center",
            "fontFamily": "Jua-Regular",
            "fontSize": 26,
          }
        }
      >
        100
      </Text>
      Ionicons
    </View>
    <RCTScrollView
      data={
        [
          "All",
          "Food",
          "Toys",
          "Clothes",
          "Other",
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      stickyHeaderIndices={[]}
      style={
        {
          "alignSelf": "center",
          "flexDirection": "row",
          "margin": 5,
          "padding": 5,
        }
      }
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#98FB98",
                "borderRadius": 10,
                "margin": 5,
                "padding": 5,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="shop-category-All"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "Jua-Regular",
                    "fontSize": 26,
                  }
                }
              >
                All
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#98FB98",
                "borderRadius": 10,
                "margin": 5,
                "padding": 5,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="shop-category-Food"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "Jua-Regular",
                    "fontSize": 26,
                  }
                }
              >
                Food
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#98FB98",
                "borderRadius": 10,
                "margin": 5,
                "padding": 5,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="shop-category-Toys"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "Jua-Regular",
                    "fontSize": 26,
                  }
                }
              >
                Toys
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#98FB98",
                "borderRadius": 10,
                "margin": 5,
                "padding": 5,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="shop-category-Clothes"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "Jua-Regular",
                    "fontSize": 26,
                  }
                }
              >
                Clothes
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#98FB98",
                "borderRadius": 10,
                "margin": 5,
                "padding": 5,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="shop-category-Other"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  {
                    "fontFamily": "Jua-Regular",
                    "fontSize": 26,
                  }
                }
              >
                Other
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <RCTScrollView
    data={
      [
        {
          "category": "Food",
          "cost": 10,
          "id": 1,
          "image": "https://via.placeholder.com/50",
          "name": "Orange Juice",
        },
        {
          "category": "Food",
          "cost": 20,
          "id": 2,
          "image": "https://via.placeholder.com/50",
          "name": "Apple Juice",
        },
        {
          "category": "Food",
          "cost": 20,
          "id": 3,
          "image": "https://via.placeholder.com/50",
          "name": "Banana Juice",
        },
        {
          "category": "Toys",
          "cost": 50,
          "id": 4,
          "image": "https://via.placeholder.com/50",
          "name": "Ball",
        },
        {
          "category": "Clothes",
          "cost": 40,
          "id": 5,
          "image": "https://via.placeholder.com/50",
          "name": "Los Diablos Ski Mask",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="shop-item-1"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#ffffff",
                "borderRadius": 10,
                "flexDirection": "row",
                "margin": 5,
                "padding": 5,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://via.placeholder.com/50",
                }
              }
              style={
                {
                  "alignSelf": "center",
                  "height": 50,
                  "width": 50,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                id="shop-item-1-name"
                style={
                  {
                    "alignSelf": "center",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginLeft": 10,
                  }
                }
              >
                Orange Juice
              </Text>
              <Text
                id="shop-item-1-cost"
                style={
                  {
                    "alignSelf": "center",
                    "fontSize": 16,
                  }
                }
              >
                10
              </Text>
            </View>
            Ionicons
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="shop-item-2"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#ffffff",
                "borderRadius": 10,
                "flexDirection": "row",
                "margin": 5,
                "padding": 5,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://via.placeholder.com/50",
                }
              }
              style={
                {
                  "alignSelf": "center",
                  "height": 50,
                  "width": 50,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                id="shop-item-2-name"
                style={
                  {
                    "alignSelf": "center",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginLeft": 10,
                  }
                }
              >
                Apple Juice
              </Text>
              <Text
                id="shop-item-2-cost"
                style={
                  {
                    "alignSelf": "center",
                    "fontSize": 16,
                  }
                }
              >
                20
              </Text>
            </View>
            Ionicons
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="shop-item-3"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#ffffff",
                "borderRadius": 10,
                "flexDirection": "row",
                "margin": 5,
                "padding": 5,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://via.placeholder.com/50",
                }
              }
              style={
                {
                  "alignSelf": "center",
                  "height": 50,
                  "width": 50,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                id="shop-item-3-name"
                style={
                  {
                    "alignSelf": "center",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginLeft": 10,
                  }
                }
              >
                Banana Juice
              </Text>
              <Text
                id="shop-item-3-cost"
                style={
                  {
                    "alignSelf": "center",
                    "fontSize": 16,
                  }
                }
              >
                20
              </Text>
            </View>
            Ionicons
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="shop-item-4"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#ffffff",
                "borderRadius": 10,
                "flexDirection": "row",
                "margin": 5,
                "padding": 5,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://via.placeholder.com/50",
                }
              }
              style={
                {
                  "alignSelf": "center",
                  "height": 50,
                  "width": 50,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                id="shop-item-4-name"
                style={
                  {
                    "alignSelf": "center",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginLeft": 10,
                  }
                }
              >
                Ball
              </Text>
              <Text
                id="shop-item-4-cost"
                style={
                  {
                    "alignSelf": "center",
                    "fontSize": 16,
                  }
                }
              >
                50
              </Text>
            </View>
            Ionicons
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="shop-item-5"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#ffffff",
                "borderRadius": 10,
                "flexDirection": "row",
                "margin": 5,
                "padding": 5,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://via.placeholder.com/50",
                }
              }
              style={
                {
                  "alignSelf": "center",
                  "height": 50,
                  "width": 50,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                id="shop-item-5-name"
                style={
                  {
                    "alignSelf": "center",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginLeft": 10,
                  }
                }
              >
                Los Diablos Ski Mask
              </Text>
              <Text
                id="shop-item-5-cost"
                style={
                  {
                    "alignSelf": "center",
                    "fontSize": 16,
                  }
                }
              >
                40
              </Text>
            </View>
            Ionicons
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    id="shop-reset-points-button"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "fontFamily": "Jua-Regular",
          "fontSize": 26,
        }
      }
    >
      Reset points
    </Text>
  </View>
</View>
`;
